csrc/mmdeploy/preprocess/transform/letter_resize.cpp

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) OpenMMLab. All rights reserved.
+
+#include <algorithm>
+#include <array>
+
+#include "mmdeploy/core/tensor.h"
+#include "mmdeploy/core/utils/formatter.h"
+#include "mmdeploy/operation/managed.h"
+#include "mmdeploy/operation/vision.h"
+#include "mmdeploy/preprocess/transform/tracer.h"
+#include "mmdeploy/preprocess/transform/transform.h"
+
+using namespace std;
+
+namespace mmdeploy::transform {
+
+class LetterResize : public Transform {
+ public:
+  explicit LetterResize(const Value& args) {
+    keep_ratio_ = args.value<bool>("keep_ratio", false);
+    if (args.contains("scale")) {
+      if (args["scale"].is_number_integer()) {
+        auto size = args["scale"].get<int>();
+        img_scale_ = {size, size};
+      } else if (args["scale"].is_array()) {
+        if (args["scale"].size() != 2) {
+          MMDEPLOY_ERROR("'scale' expects an array of size 2, but got {}", args["scale"].size());
+          throw std::length_error("'scale' expects an array of size 2");
+        }
+        auto height = args["scale"][0].get<int>();
+        auto width = args["scale"][1].get<int>();
+        img_scale_ = {height, width};
+      } else {
+        MMDEPLOY_ERROR("'scale' is expected to be an integer or and array of size 2");
+        throw std::domain_error("'scale' is expected to be an integer or and array of size 2");
+      }
+    }
+    if (args.contains("pad_val")) {
+      if (args["pad_val"].is_number()) {
+        pad_val_ = args["pad_val"].get<float>();
+      } else if (args["pad_val"].contains("img")) {
+        pad_val_ = args["pad_val"]["img"].get<float>();
+      }
+    }
+    interpolation_ = args.value<string>("interpolation", "bilinear");
+    allow_scale_up_ = args.value<bool>("allow_scale_up", true);
+    use_mini_pad_ = args.value<bool>("use_mini_pad", false);
+    stretch_only_ = args.value<bool>("stretch_only", false);
+
+    vector<string> interpolations{"nearest", "bilinear", "bicubic", "area", "lanczos"};
+    if (std::find(interpolations.begin(), interpolations.end(), interpolation_) ==
+        interpolations.end()) {
+      MMDEPLOY_ERROR("'{}' interpolation is not supported", interpolation_);
+      throw std::invalid_argument("unexpected interpolation");
```

Comment:
Let's use throw_exception(eInvalidArgument); instead of `throw std::invalid_argument`